@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -702,18 +717,9 @@ export class TextElementsRenderer {
     private reset() {
         this.m_cameraLookAt.copy(this.m_viewState.lookAtVector);
         this.m_screenCollisions.reset();
-        for (const textRenderer of this.m_textRenderers) {
-            textRenderer.textCanvas.clear();
-            textRenderer.poiRenderer.reset();
-        }
-    }
-
-    /**
-     * Update state at the end of a frame.
-     */
-    private updateTextRenderers() {
-        for (const textRenderer of this.m_textRenderers) {
-            textRenderer.poiRenderer.update();
+        for (const screenSpaceRenderer of this.m_screenSpaceRenderers) {
+            screenSpaceRenderer.textCanvas?.clear();
+            screenSpaceRenderer.poiRenderer?.reset();
```

Comment:
We're clearing here the same poiRenderer multiple times.

Reply:
resolved